Listenbrainz/UI/Screens/Shazam/PlayerView.swift

```diff
@@ -6,132 +6,175 @@
 //
 
 
+
 import SwiftUI
 import ShazamKit
 
-//extension SHMatchedMediaItem: Identifiable {
-//
-//    public var id: String {
-//        shazamID ?? ""
-//    }
-//}
-
 struct PlayerView: View {
-    @State private var fileName = ""
-    @State private var openFile = false
-    @StateObject private var viewModel: ShazamViewModel = .init()
-    @State private var animateShazamButton = false
-    
-    var body: some View {
-        content
-            .sheet(item: $viewModel.mediaItem) { mediaItem in
-                MediaItemView(mediaItem: mediaItem)
-            }
-            .alert(isPresented: $viewModel.hasError) {
-                Alert(title: Text("Error"), message: Text(viewModel.error?.localizedDescription ?? ""))
-            }
-    }
-    
-    @ViewBuilder
-    var content: some View {
-        if viewModel.matching {
-            ProgressView("Listening...")
-                .tint(Color.primary)
-                .foregroundColor(Color.secondary)
+  @State private var fileName = ""
+  @State private var openFile = false
+  @StateObject private var shazamViewModel: ShazamViewModel = .init()
+  @State private var animateShazamButton = false
+  @StateObject var viewModel: YIMViewModel
+
+  var body: some View {
+    content
+      .sheet(item: $shazamViewModel.mediaItem) { mediaItem in
+        MediaItemView(mediaItem: mediaItem)
+      }
+      .alert(isPresented: $shazamViewModel.hasError) {
+        Alert(title: Text("Error"), message: Text(shazamViewModel.error?.localizedDescription ?? ""))
+      }
+  }
+
+  @ViewBuilder
+  var content: some View {
+    if shazamViewModel.matching {
```

Comment:
We should move this to a separate component

Reply:
@akshaaatt we have to make seperate bottom toolbar item for shazam right?

Reply:
Yes, @gauravbhardwaj7

Reply:
done